pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -1078,17 +1079,42 @@ func getInfinispanConditions(pods []corev1.Pod, m *infinispanv1.Infinispan, clus
 	sort.Strings(views)
 	if len(errors) == 0 {
 		if len(views) == 1 {
-			wellformed.Status = metav1.ConditionTrue
-			wellformed.Message = "View: " + views[0]
+			wellFormed.Status = metav1.ConditionTrue
+			wellFormed.Message = "View: " + views[0]
+
+			// Perform cross-site view validation
+			if m.HasSites() {
+				crossSiteFormed := infinispanv1.InfinispanCondition{Type: infinispanv1.ConditionCrossSiteFormed, Status: metav1.ConditionTrue}
+				sitesView, err := cluster.GetSitesView(pods[0].Name)
+				if err == nil {
+					locations := m.GetSitesLocations("")
+					sites := make([]string, len(locations))
+					for i, location := range locations {
+						sites[i] = location.Name
+						if !sitesView[location.Name] {
+							crossSiteFormed.Status = metav1.ConditionFalse
+							crossSiteFormed.Message = fmt.Sprintf("Site '%s' not ready", location.Name)
```

Comment:
@dmvolod When we say the site is not ready, is it that the remote cluster is not fully started and the user just needs to wait? Or could there potentially be an error?

Reply:
I'm not sure we have an additional info for this except situation we discussing with @pruivo privately now

Reply:
(sorry to revive this)
we don't have info about the cause. It just means the remote site isn't connected to this site. It doesn't know if the remote site is started, or there is a connection error or a configuration error (no way t know)

@dmvolod is `pods[0]` in `cluster.GetSitesView(pods[0].Name)` the coordinator?

Reply:
@pruivo no, it's just first Pod, but we can choose a Pod with `coordinator` label also. However `GetSitesView` works fine with any Pod, probably I test a single Pod deployment, need to investigate more.

Reply:
please choose the coordinator, it is the one with the real xsite view.
the other pods may have a partial xsite view or no xsite view at all.

Reply:
Yes, sure. Will fix, thanks.